sdk/search/azure-search-documents/azure/search/documents/_service/_search_service_client.py

```diff
@@ -411,3 +413,128 @@ def create_or_update_synonym_map(self, name, synonyms, **kwargs):
+
+    @distributed_trace
+    def get_skillset(self, name, **kwargs):
+        # type: (str, **Any) -> dict
+        """Retrieve a named Skillset in an Azure Search service
+
+        :param name: The name of the Skillset to get
+        :type name: str
+        :return: The retrieved Skillset
+        :rtype: dict
+        :raises: :class:`~azure.core.exceptions.ResourceNotFoundError`
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/sample_skillset_operations.py
+                :start-after: [START get_skillset]
+                :end-before: [END get_skillset]
+                :language: python
+                :dedent: 4
+                :caption: Get a Skillset
+
+        """
+        kwargs["headers"] = self._merge_client_headers(kwargs.get("headers"))
+        result = self._client.skillsets.get(name, **kwargs)
+        return skillset_as_dict(result)
+
+    @distributed_trace
+    def delete_skillset(self, name, **kwargs):
+        # type: (str, **Any) -> None
+        """Delete a named Skillset in an Azure Search service
+
+        :param name: The name of the Skillset to delete
+        :type name: str
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/sample_skillset_operations.py
+                :start-after: [START delete_skillset]
+                :end-before: [END delete_skillset]
+                :language: python
+                :dedent: 4
+                :caption: Delete a Skillset
+
+        """
+        kwargs["headers"] = self._merge_client_headers(kwargs.get("headers"))
+        self._client.skillsets.delete(name, **kwargs)
+
+    @distributed_trace
+    def create_skillset(self, name, skills, description, **kwargs):
+        # type: (str, Sequence[Skill], str, **Any) -> dict
+        """Create a new Skillset in an Azure Search service
+
+        :param name: The name of the Skillset to create
+        :type name: str
+        :param skills: A list of Skill objects to include in the Skillset
+        :type skills: List[Skill]]
+        :param description: A description for the Skillset
+        :type description: Optional[str]
+        :return: The created Skillset
+        :rtype: dict
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/sample_skillset_operations.py
+                :start-after: [START create_skillset]
+                :end-before: [END create_skillset]
+                :language: python
+                :dedent: 4
+                :caption: Create a Skillset
+
+        """
+        kwargs["headers"] = self._merge_client_headers(kwargs.get("headers"))
+
+        skillset = Skillset(name=name, skills=list(skills), description=description)
+
+        result = self._client.skillsets.create(skillset, **kwargs)
+        return skillset_as_dict(result)
+
+    @distributed_trace
+    def create_or_update_skillset(self, name, skills, description, **kwargs):
+        # type: (str, Sequence[Skill], str, **Any) -> dict
```

Comment:
Do we want to support match conditions?

Reply:
And, do we support partial updates?

Reply:
Probably but I was going to add a separate issue to add those in a different PR

Reply:
No partial updates supported AFAIK, skills list is always replaced

Reply:
Always replacing is the behavior of rest api. What's our designed behavior? e.g. if I only want to modify description, do I have to get the entire skills and put it here? Can I just leave it None to indicate I don't want to modify skills? @johanste

Reply:
No, the service pukes:
```
ValidationError: Parameter 'Skillset.skills' can not be None.
```

Reply:
I mean I call
create_or_update_skillset(name='name', skills=None, description='new description')
and our SDK does:
skillset = get_skillset(name)
skillset.description=new description
create_or_update_skillset(name, skillset, skillset.description)

Reply:
No, the return is a dict:
```
{'name': 'foo',
 'skills': [<azure.search.documents._service._generated.models._models_py3.EntityRecognitionSkill at 0x110d29940>],
 'description': 'desc',
 'e_tag': '"0x8D7E19BA6ECB1A3"'}
```
and `create_or_update_skillset(**skillset)` barfs because of the unexpected `e_tag`. We could certainly do ugly arg handling things to make it possible to pass a dict back in but I have not heard anyone suggest this is common usage that is important to support.

Reply:
We could of course just switch to accepting/returning the real models instead of flattening the args and returning a dict. cc @johanste

Reply:
One of the problems we have here is that there are several different user intents that we are trying to map into a single function - Create, partial or full Update and Replace.

For Create and Replace, flattening the parameters make sense. For update, you have update-based-on-resource-state (I need to look at the state of the resource before I determine if and what updates I want to make) and partial updates (I just want to change this subset of properties on the resource).

It sounds like the only thing the service supports is effectively the "replace" (not partial update) scenario. This makes it much more likely to, as @xiangyan99 points out, the user having to get the resource, modify it locally and then put it back (the service didn't allow me to send just the value I wanted to mutate). 

In these cases, the best option is often to take both the full model and the flattened parameters. Any flattened value would override what's in the model. Yes, you can specify values twice (in the model and as a direct parameter). Which is somewhat problematic. But it generally works. And is relatively easy to explain.

Reply:
@johanste just to make sure I understand you are proposing to move all the named params to `kwargs` and also add a `skillset` kwarg that can be passed?
```
def create_or_update_skillset(self, **kw):
    if "skillset" in kw:
        skillset = kw.pop('skillset')
        for param in ["name", "description", "skills"]:
            if param in kw: 
                setattr(skillset, param, kw.pop(param))
    else:
        # all of these are *required* to create a new Skillset
        for param in ["name", "description", "skills"]
            if param not in kw:
                raise ValueError()
        skillset = Skillset(name=kw.pop("name"), 
                            description=kw.pop("description"), 
                            skills=kw.pop("skills"))

I'm not a huge fan of this, but will update if you confirm this is what you had in mind.

Reply:
Also, would need to change `get_skillet` to return an actual `Skillset` and not a dict (otherwise there would not be a ready source of obtaining a `Skillset` to modify for update in the first place)

Reply:
It also seems like modifying the passed-in `Skillset` would be surprising to users, so I guess
```
Skillset.deserialize(skillset.serialize())
``` 
unless there is some better way?

Reply:
@johanste @xiangyan99 please see the latest two commits